### PR TITLE
feat: port nodejs tests to typescript

### PR DESCRIPTION
**Pull request by Mause**
CC. @antonycourtney @jwills @hamilton 

The intention here is a pure port - not trying to improve the cleanliness yet at all

### CODE REVIEW

File: tools/nodejs/lib/duckdb.d.ts [hunk 1]
```diff
@@ -1,103 +1,120 @@
-/**
```
Comment (Mause) [inline tools/nodejs/lib/duckdb.d.ts:1]: Looks like the line endings got flipped? Not sure if it's worth correcting

**Review by Mytherin — APPROVED** [review]
LGTM - maybe someone that is more knowledgeable about the node client can also have a look?

**Comment by Mause** [comment]
> LGTM - maybe someone that is more knowledgeable about the node client can also have a look?

Yep, the three people I tagged in the description offered to review, though I can't add them as formal reviewers

**Comment by Mytherin** [comment]
@antonycourtney could you perhaps have a look at this when you have time? If it looks good to you I think this is ready to merge.

File: tools/nodejs/test/interrupt.test.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-var assert = require('assert');
+import * as sqlite3 from '..';
+import * as assert from 'assert';
+import {DuckDbError} from "..";
 
 // FIXME each is not streaming yet
-return
 
-describe('interrupt', function() {
+describe.skip('interrupt', function() {
```
Comment (antonycourtney) [inline tools/nodejs/test/interrupt.test.ts:7]: Similar comment to earlier. Do we have to skip?  Running at least exercises the code path and ensures it is not crashing. (Obviously adding some kind of assertion would be preferable though).

File: tools/nodejs/test/jsdoc.test.ts [hunk 3]
```diff
@@ -1,60 +1,59 @@
...
-    it(clazz, () => {
-      checkDocs(duckdb[clazz].prototype, `module:duckdb~${clazz}`);
-    });
-  }
-
-  it('module root', () => {
-    checkDocs(duckdb, 'module:duckdb');
-  });
-});
+/**
```
Comment (antonycourtney) [inline tools/nodejs/test/jsdoc.test.ts:1]: What's going on with this file?  line endings again?

**Review by antonycourtney — APPROVED** [review]
Did a full pass. Looks great, thanks so much for doing this!  Almost all the changes appear to be addition of type signatures (as expected).  A couple of questions about some tests now being skipped; ideally we shouldn't skip them.

File: tools/nodejs/lib/duckdb.d.ts [hunk 4]
```diff
@@ -1,103 +1,120 @@
...
+
+  interrupt(): void;
+
+  register_buffer(name: string, array: ArrayLike<any>, force: boolean, callback?: Callback<void>): void;
+
+  unregister_buffer(name: string, callback?: Callback<void>): void;
+}
+
+export class Statement {
+  sql: string;
```
[the remark this answers is not in the document]
Comment (Mause) [reply tools/nodejs/lib/duckdb.d.ts:92]: Yep, it should be. I'll update the constructor

File: tools/nodejs/test/affected.test.ts [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
                 assert.equal(j++, this.lastID);
             });
         }
         db.wait(done);
     });
 
 
-    it('should return the correct changes count', function(done) {
-        db.run("UPDATE foo SET id = id + 1 WHERE id % 2 = 0", function(err) {
+    it.skip('should return the correct changes count', function(done) {
```
[the remark this answers is not in the document]
Comment (Mause) [reply tools/nodejs/test/affected.test.ts:27]: Yeah, it was also previously commented out, hence why I'm skipping it. I'll see what it takes to get it to pass, or run at least without the assertions

File: tools/nodejs/test/jsdoc.test.ts (hunk 3 above)
Comment (Mause) [reply tools/nodejs/test/jsdoc.test.ts:1]: Yep, looks like it

File: tools/nodejs/test/interrupt.test.ts (hunk 2 above)
Comment (Mause) [reply tools/nodejs/test/interrupt.test.ts:7]: These tests weren't running at all previously, I'll see what's required to get them passing

File: tools/nodejs/test/affected.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (antonycourtney) [reply tools/nodejs/test/affected.test.ts:27]: Ah, sorry, I missed the open comment when I read the diff. In that case: yeah, skipping seems totally fine, I just didn't want to lose any existing code coverage.

File: tools/nodejs/test/interrupt.test.ts (hunk 2 above)
Comment (antonycourtney) [reply tools/nodejs/test/interrupt.test.ts:7]: Again, sorry I missed if this was commented out!  No need to get it running in that case; best to just get this PR landed.

**Review by antonycourtney — APPROVED** [review]
Looks great!  Low risk and strictly an improvement.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**